.github/workflows/lint.yml

```diff
@@ -1,44 +1,72 @@
 name: Lint
 
-on: [push, pull_request]
+on:
+  push:
+  pull_request:
+
+permissions:
+  contents: read
 
 jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
-    - name: Cancel previous run
-      uses: styfle/cancel-workflow-action@0.9.1
-      with:
-        access_token: ${{ github.token }}
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.8
-      uses: actions/setup-python@v2
-      with:
-        python-version: 3.8
-    - name: Upgrade pip
-      run: |
-        python -m pip install --upgrade pip
-    - name: Install requirements
-      run: |
-        python -m pip install -r requirements.txt
-    - name: Install dependencies
-      run: |
-        python -m pip install -e .
-    - name: flake8
-      run: |
-        make flake8
-    - name: isort and yapf
-      run: |
-        make py-format
-    - name: addlicense
-      run: |
-        make addlicense
-    - name: mypy
-      run: |
-        make mypy
+      - name: Cancel previous run
+        uses: styfle/cancel-workflow-action@0.10.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: "recursive"
+          fetch-depth: 1
+
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+
+      - name: Setup CUDA Toolkit
+        uses: Jimver/cuda-toolkit@v0.2.7
+        id: cuda-toolkit
+        with:
+          cuda: "11.7.0"
+          method: network
+          sub-packages: '["nvcc"]'
+      - run: |
+          echo "Installed CUDA version is: ${{steps.cuda-toolkit.outputs.cuda}}"
+          echo "CUDA install location: ${{steps.cuda-toolkit.outputs.CUDA_PATH}}"
+          nvcc -V
+
+      - name: Upgrade pip
+        run: |
+          python -m pip install --upgrade pip setuptools
+
+      - name: Install TorchOpt
+        run: |
+          python -m pip install -e .
+
+      - name: flake8
+        run: |
+          make flake8
+
+      - name: isort and yapf
+        run: |
+          make py-format
+
+      - name: addlicense
+        run: |
+          make addlicense
+
+      - name: mypy
+        run: |
+          make mypy
+
     # - name: docstyle
```

Comment:
If you got time, maybe add the docstyle? (docsting is a mess right now I think)

Reply:
We may open a new PR for this.